Gems/PhysX/Code/Source/PhysXCharacters/Components/RagdollComponent.cpp

```diff
@@ -389,17 +406,63 @@ namespace PhysX
             }
         }
 
+        // If mass ratio clamping is enabled, iterate out from the root and clamp mass values
+        if (m_enableMassRatioClamping)
+        {
+            const float maxMassRatio = AZStd::GetMax(1.0f + AZ::Constants::FloatEpsilon, m_maxMassRatio);
+
+            // figure out the depth of each node in the tree, so that nodes can be visited from the root outwards
+            AZStd::vector<Utils::Characters::DepthData> nodeDepths =
+                Utils::Characters::ComputeHierarchyDepths(ragdollConfiguration.m_parentIndices);
+
+            AZStd::sort(
+                nodeDepths.begin(), nodeDepths.end(),
+                [](const Utils::Characters::DepthData& d1, const Utils::Characters::DepthData& d2)
+                {
+                    return d1.m_depth < d2.m_depth;
+                });
```

Comment:
question: Would it be worth it to have this sort in the ```Utils::Characters::ComputeHierarchyDepths``` function? or is it expected not to always need to sort?

Reply:
Yeah, that's worth thinking about. I think we should probably have a refactor of CreateRagdoll, it's over 100 lines long now, but I'd suggest we do that in another change to keep things separate. I've opened https://github.com/o3de/o3de/issues/5698 to keep track of that work.